spec/providers/coreIntegration/tcpsocket.integration.src.js

```diff
@@ -100,5 +83,22 @@ module.exports = function (provider, setup) {
         done();
       });
   });
+
+  it("Works as a Client", function (done) {
+    socket.connect('www.google.com', 80, function () {
+      console.warn('connected');
+      var written = false;
+      socket.write(rawStringToBuffer('GET / HTTP/1.0\n\n'), function (okay) {
+        console.warn('written');
+        written = true;
+      });
+      setTimeout(function () {
```

Comment:
Good points and I agree with cleaning up this test, but I'm not sure it will fix the issue. I tried paring down the test (in its old run-first position) bit by bit, until eventually I just tried this instead:

it("works", function (done) {
    done();
});

And for some reason I've not fully fathomed it caused this error in the following test:
1) integration: core.tcpsocket Gives error when connecting to invalid domain
    Message:
      Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.


    Stacktrace:
      Error: Timeout - Async callback was not invoked within timeout specified by jasmine.DEFAULT_TIMEOUT_INTERVAL.
    at Timer.listOnTimeout [as ontimeout] (timers.js:112:15)

This later test doesn't use setTimeout, it's hitting the actual jasmine timeout. Removing the test entirely lets the last test pass, as does moving it down.

Anyway, I'll remove the setTimeouts anyway as that makes sense, but still suspect there's some other strangeness here.